src/index.js

```diff
@@ -1,98 +1,27 @@
-    data.sources.map(({ id }, index) => {
-      const selectOptions = document.createElement("option");
-      selectOptions[index] += selectOptions.text = id;
-      sourceSelectBox.appendChild(selectOptions);
-    });
-  }
-}
-class NewssourceSelectComponent extends ApiFetcher {
-  constructor(data) {
-    super();
-  }
-  async fetch(httpurl) {
-    return await super.fetch(httpurl);
-  }
-  render(data) {
-    const elementId = document.getElementById("newsDetails");
-    let returnHtml = "",
-      uniqueVal = [];
-    if (data.status === "error") {
-      returnHtml = `<div id="error">${data.message}</div>`;
-    } else {
-      data.articles.map(
-        (
-          { author, title, description, publishedAt, url, urlToImage },
-          index
-        ) => {
-          uniqueVal = index === 0 ? `<h1>${author}</h1>` : "";
-          returnHtml += `${uniqueVal}<div class="newsTitle">${title}</div><div class="newsDescription">${description}</div><div class="publishDate">${publishedAt}</div><div class="imageContainer"><a href="${url}" target="_blank"><img class ="imageLazy" data-lazy=${urlToImage} /></a></div>`;
-        }
-      );
-    }
-    elementId.innerHTML = returnHtml;
-  }
-}
-class NewsAppConent {
-  constructor() {
-    this.newssourceComponentClass = new NewssourceComponent(this);
-    this.newssourceSelectComponent = new NewssourceSelectComponent(this);
-  }
-  intialize() {
-    const sourceSelectBox = document.getElementById("newsSource");
-
-    sourceSelectBox.addEventListener("click", event => {
-      urlConstructor.newsCategory = event.target.value;
-    });
+import urlConstructor from "./config.json";
+// import ApiCall from "./apiRequest.js";
+import Api from './sourceFetch'
 
-    const newsBtn = document.getElementById("newsSourceBtn");
-    newsBtn.addEventListener("click", () => {
-      urlConstructor.urlConstructor.newsCategory = sourceSelectBox.value;
-      let url =
-        urlConstructor.urlConstructor.url +
-        urlConstructor.urlConstructor.newsCategory +
-        urlConstructor.urlConstructor.apiKey;
+let baseUrl = urlConstructor.urlConstructor[0].apiURL;
+let apiKey = urlConstructor.urlConstructor[1].apiKey
+// let loadUrl = "https://newsapi.org/v1/sources?apiKey=720c0314e8b2423eb7e1ffca5a1eeeb1"
+console.log(baseUrl, apiKey);
 
-      this.newssourceSelectComponent.fetch(url).then(value => {
-        this.newssourceSelectComponent.render(value);
-      });
-      import(/* webpackChunkName: "print" */ "./print.js").then(mod => {
-        console.log(mod);
-      });
-    });
-    this.newsSource = this.newssourceComponentClass.fetch().then(value => {
-      this.newssourceComponentClass.render(value);
-    });
+const handler = {
+  get: function(target, name) {
+    console.log(`calling: ${name}`);
+    return target[name]
   }
-}
-const requestCall = new ApiFetcher();
-const intializeFunctions = new NewsAppConent();
+};
 
-intializeFunctions.intialize();
+const api = new Api(baseUrl, apiKey);
```

Comment:
why you are creating two instances with the same params?

Reply:
there are two requests 
1). Source call for onload constructing Selectbox.
2). Article call on selection selectBox dropdown Displaying the new content.